cocotb/types/logic_array.py

```diff
@@ -189,6 +201,31 @@ def is_resolvable(self) -> bool:
 
     @property
     def integer(self) -> int:
+        """
+        Raises ValueError if the values cannot be converted to binary ('X' or 'Z')
+        """
+        return int(self)
+
+    def __eq__(self, other):
+        """
+        Overrides the default implementation to enable comparision
+        with integers and strings (case insensitive)
+        """
+        if isinstance(other, int):
+            return self.integer == other
+        if isinstance(other, str):
+            # case insensitive to make sure 'x' == 'X' and 'z' == 'Z'
+            # internally LogicArray uses upper case but use may not know about it
+            return self.binstr == other.upper()
```

Comment:
And in fact, this could be collapsed into the previous statement by using a tuple in the `isinstance` check.
```suggestion
            return self == LogicArray(other)
```